design/unnumbered-bgp.md

```diff
@@ -0,0 +1,498 @@
+
+IPv4 Unicast Summary (VRF default):
+BGP router identifier 11.11.11.254, local AS number 65004 vrf-id 0
+BGP table version 4
+RIB entries 3, using 288 bytes of memory
+Peers 3, using 39 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+kind-control-plane(eth00) 4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+kind-worker(eth01)        4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+kind-worker2(eth02)       4      65000       510       510        4    0    0 00:24:43            1        2 k8s-node
+
+Total number of neighbors 3
+
+IPv6 Unicast Summary (VRF default):
+BGP router identifier 11.11.11.254, local AS number 65004 vrf-id 0
+BGP table version 3
+RIB entries 1, using 96 bytes of memory
+Peers 3, using 39 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+kind-control-plane(eth00) 4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker(eth01)        4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+kind-worker2(eth02)       4      65000       510       510        3    0    0 00:24:43            1        1 k8s-node
+
+
+> k -n metallb-system exec -it -c frr frr-k8s-daemon-kzpz2 -- vtysh -c 'show bgp summary'
+
+IPv4 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 2
+RIB entries 3, using 288 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        2    0    0 00:25:15            1        2 TOR
+
+Total number of neighbors 1
+
+IPv6 Unicast Summary (VRF default):
+BGP router identifier 172.19.0.3, local AS number 65000 vrf-id 0
+BGP table version 1
+RIB entries 1, using 96 bytes of memory
+Peers 1, using 13 KiB of memory
+Peer groups 1, using 64 bytes of memory
+
+Neighbor                  V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
+unnumbered-p2p-peer(net0) 4      65004       512       512        1    0    0 00:25:15            0        1 TOR
+
+Total number of neighbors 1
+```
+### Topics
+
+#### Loopback IPv4 address
+
+Linux sends and receives IPv4 packets over an interface that does not have an IPv4 address.
+It does not look we need to add address in the loopback.
+
+#### RA enabled on both ends of the link
+
+According to the design of unnumber BGP peering, we need to enable RAs.
+
+#### BGP RouterID
+
+//TODO is there a requirement for that?
```

Comment:
wdym here?

Reply:
I have a lack of  BGP knowledge what is the general impact of this RouterID in our use cases?  Do you know?

Reply:
frr sets the router id with the lowest ip of the interfaces. So far, we didn't have a need to do otherwise. If this is not relevant please remove the section as it's distracting.

Reply:
I was expected if you could explain me how the RouterID is important or not. Example when we have single IPv6 stack. But I have that for discussion, removed